### PR TITLE
Add more validation in lifecycle configuration class

### PR DESCRIPTION
**Pull request by balamurugana**

### CODE REVIEW

File: api/src/main/java/io/minio/messages/LifecycleRule.java [hunk 1]
```diff
@@ -55,13 +55,24 @@ public LifecycleRule(
...
       @Nullable @Element(name = "Transition", required = false) Transition transition) {
+    if (abortIncompleteMultipartUpload == null
+        && expiration == null
+        && noncurrentVersionExpiration == null
+        && noncurrentVersionTransition == null
+        && transition == null) {
+      throw new IllegalArgumentException(
+          "At least one of action (AbortIncompleteMultipartUpload, Expiration, "
+              + "NoncurrentVersionExpiration, NoncurrentVersionTransition or Transition) must to "
+              + "be specified in a rule");
```
Comment (anjalshireesh) [inline api/src/main/java/io/minio/messages/LifecycleRule.java:73]:
```suggestion
              + "NoncurrentVersionExpiration, NoncurrentVersionTransition or Transition) must "
              + "be specified in a rule");
```

**Review by anjalshireesh — APPROVED** [review]
LGTM, one minior comment

File: api/src/main/java/io/minio/messages/LifecycleRule.java (hunk 1 above)
Comment (balamurugana) [reply api/src/main/java/io/minio/messages/LifecycleRule.java]: Done

**Review by Praveenrajmani — APPROVED** [review]

**Review by anjalshireesh — APPROVED** [review]
LGTM

**Pull request merged**